### PR TITLE
[CT-1040] Push Notifications (2 of 4) - Add Token To Postgres 

### PR DESCRIPTION
**Pull request by adamfraser**
Adds functionality to store Tokens in Postgres

### Changelist
[Describe or list the changes made in this PR]

## Summary by CodeRabbit

- **New Features**
	- Introduced a system for managing Firebase notification tokens within a PostgreSQL database.
	- Added support for creating, updating, and deleting notification tokens.
	- Enhanced data seeding capabilities for Firebase notification tokens.

- **Documentation**
	- Enhanced type definitions for better clarity and safety when handling notification tokens.

- **Tests**
	- Implemented comprehensive unit tests for the Firebase notification token management functionalities.

### CODE REVIEW

File: indexer/packages/postgres/src/types/db-model-types.ts [hunk 1]
```diff
@@ -295,6 +295,13 @@ export interface AffiliateReferredUserFromDatabase {
   referredAtBlock: string,
 }
 
+export interface FirebaseNotificationTokenFromDatabase {
+  address: WalletFromDatabase['address'],
```
Comment (dydxwill) [inline indexer/packages/postgres/src/types/db-model-types.ts:299]: can just do `address: string` here

File: indexer/packages/postgres/src/types/firebase-notification-token-types.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  token: string,
+  address: string,
+  language: string,
+  updatedAt: IsoString,
+}
+
+export interface FirebaseNotificationTokenUpdateObject {
+  token: string,
+  address: string,
+  language: string,
```
Comment (dydxwill) [inline indexer/packages/postgres/src/types/firebase-notification-token-types.ts:15]: do all these fields need to be present in the update object? if not, can we use `?:`

File: indexer/packages/postgres/src/models/firebase-notification-token-model.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
+import { Model } from 'objection';
```
Comment (dydxwill) [inline indexer/packages/postgres/src/models/firebase-notification-token-model.ts:1]: can this file follow the existing convention of other model files in terms of ordering, `jsonSchema`, and `sqlToJsonConversions`, `tableName`, idColumn`?

File: indexer/packages/postgres/src/db/migrations/migration_files/20240809153326_create_firebase_notification_tokens_table.ts [hunk 4]
```diff
@@ -0,0 +1,16 @@
+import * as Knex from 'knex';
+
+export async function up(knex: Knex): Promise<void> {
+  return knex.schema.createTable('firebase_notification_tokens', (table) => {
+    table.increments('id').primary();
+    table.string('token').notNullable().unique();
+    table.string('address').notNullable();
+    table.foreign('address').references('wallets.address').onDelete('CASCADE');
```
Comment (dydxwill) [inline indexer/packages/postgres/src/db/migrations/migration_files/20240809153326_create_firebase_notification_tokens_table.ts:8]: I'm not sure the `onDelete('CASCADE')` is needed here. It doesn't seem to be present on other foreign keys defined in other tables.

Comment (adamfraser) [reply indexer/packages/postgres/src/db/migrations/migration_files/20240809153326_create_firebase_notification_tokens_table.ts:8]: This is basically saying that if we delete an wallet address the DB should automatically delete any tokens that reference that address. Which is the behavior we want imo.

File: indexer/packages/postgres/src/models/firebase-notification-token-model.ts (hunk 3 above)
Comment (dydxwill) [reply indexer/packages/postgres/src/models/firebase-notification-token-model.ts:1]: Still doesn't match other files in term of ordering etc. 

for example, `asset-model.ts` has following structure: ```
export default class AssetModel extends Model {
  static get tableName() {
  }

  static get idColumn() {
  }

  static relationMappings = {};

  static get jsonSchema() {}

  static get sqlToJsonConversions() {}

  id!: string;

  symbol!: string;

  atomicResolution!: number;

  hasMarket!: boolean;

  marketId?: number;
}
```

**Review by dydxwill — APPROVED** [review]

**Review by dydxwill — APPROVED** [review]

**Pull request merged**